collector/lib/CollectorConfig.cpp

```diff
@@ -400,6 +405,61 @@ void CollectorConfig::HandleSinspEnvVars() {
   }
 }
 
+bool CollectorConfig::YamlConfigToConfig(YAML::Node& yamlConfig) {
+  if (yamlConfig.IsNull() || !yamlConfig.IsDefined()) {
+    CLOG(FATAL) << "Unable to read config from config file";
+    return false;
+  }
+  YAML::Node networkConnectionConfig = yamlConfig["networkConnectionConfig"];
+  if (!networkConnectionConfig) {
+    CLOG(WARNING) << "No networkConnectionConfig in config file";
+    return false;
+  }
+
+  bool enableExternalIps = false;
+  try {
+    if (networkConnectionConfig["enableExternalIps"]) {
+      enableExternalIps = networkConnectionConfig["enableExternalIps"].as<bool>();
+    }
+  } catch (const std::exception& e) {
+    CLOG(FATAL) << "An error occurred getting enableExternalIps";
+  }
```

Comment:
You should be able to directly provide a default here
```suggestion
  if (networkConnectionConfig["enableExternalIps"]) {
    enableExternalIps = networkConnectionConfig["enableExternalIps"].as<bool>(false);
  }
```